discovery-provider/src/tasks/index_nethermind.py

```diff
@@ -321,772 +231,560 @@ def create_and_raise_indexing_error(err, redis):
 
-    update_task.entity_manager_contract = entity_manager_contract
-
-    # Update redis cache for health check queries
-    update_latest_block_redis(final_poa_block)
-
-    DEFAULT_LOCK_TIMEOUT = 60 * 10  # ten minutes
-
-    # Define lock acquired boolean
-    have_lock = False
-    # Define redis lock object
-    # blocking_timeout is duration it waits to try to acquire lock
-    # timeout is the duration the lock is held
-    update_lock = redis.lock(
-        "disc_prov_lock_nethermind", blocking_timeout=25, timeout=DEFAULT_LOCK_TIMEOUT
+    rebuild_playlist_index = rebuild_playlist_index or bool(revert_playlist_entries)
+    rebuild_track_index = rebuild_track_index or bool(revert_track_entries)
+    rebuild_user_index = rebuild_user_index or bool(revert_user_entries)
+    logger.info(
+        f"Reverted {revert_block_number} in {datetime.now() - start_time} seconds"
     )
-    try:
-        # Attempt to acquire lock - do not block if unable to acquire
-        have_lock = update_lock.acquire(blocking=False)
-        if have_lock:
-            logger.set_context("request_id", self.request.id)
-            logger.info("Acquired disc_prov_lock_nethermind")
-
-            latest_block = get_latest_block(db, final_poa_block)
-
-            # Capture block information between latest and target block hash
-            index_blocks_list = []
-
-            # Capture outdated block information given current database state
-            revert_blocks_list = []
-
-            with db.scoped_session() as session:
-                block_intersection_found = False
-                intersect_block_hash = web3.toHex(latest_block.hash)
-
-                # First, we capture the block hash at which the current tail
-                # and our indexed data intersect
-                while not block_intersection_found:
-                    current_hash = web3.toHex(latest_block.hash)
-                    parent_hash = web3.toHex(latest_block.parentHash)
-
-                    latest_block_db_query = session.query(Block).filter(
-                        Block.blockhash == current_hash
-                        and Block.parenthash == parent_hash
-                        and Block.is_current == True
-                    )
-
-                    # Exit loop if we are up to date
-                    if latest_block_db_query.count() > 0:
-                        block_intersection_found = True
-                        intersect_block_hash = current_hash
-                        continue
 
-                    index_blocks_list.append(latest_block)
 
-                    parent_block_query = session.query(Block).filter(
-                        Block.blockhash == parent_hash
-                    )
-
-                    # Intersection is considered found if current block parenthash is
-                    # present in Blocks table
-                    block_intersection_found = parent_block_query.count() > 0
+@celery.task(name="index_nethermind", bind=True)
```

Comment:
might as well remove nethermind now?

Reply:
was going to do this as follow up PR but agree! it needs to be distinguished from solana so
show though. perhaps index acdc